### DISCUSSION TITLE
Release Workflow

### DISCUSSION THREAD

**Discussion by kheina** [kheina-com/Blue-Blocker#258] [General]
ok I have confirmation that both firefox and chrome work well enough that there are v0.3.9 packages uploaded to both webstores waiting for review. however the firefox one was using v1.0.0 rather than v1.0.5 and I think they added the release notes and approval notes since v1.

so all I'm saying is when yall release the next version, yell at me to check the stores at the data is correct so we can call this done

**Comment by kheina** [comment]
@MrAwesome @cooljeanius @rougetimelord @jaredkotoff reminder that in order to push a new version to web stores you must make a [release](https://github.com/kheina-com/Blue-Blocker/releases)

also we should probably figure out how to add edge to the release workflow (eventually)

**Reply by cooljeanius** [reply]
> also we should probably figure out how to add edge to the release workflow (eventually)

...and presumably Safari, too, if/when we ever fix #197...

**Reply by rougetimelord** [reply]
I didn't add a release for 0.4.0 because we still had a PR open, getting on it for 0.4.1 rn

**Reply by rougetimelord** [reply]
> ...and presumably Safari, too, if/when we ever fix https://github.com/kheina-com/Blue-Blocker/issues/197...

I sure don't want to buy a mac and xcode access and ............

**Comment by kheina** [comment]
well, I guess triggering a new version cancels the review of the previous version, so 0.3.9 won't be released on firefox. also release notes and approver notes still don't work on the latest version of that github action. maybe someone else can find one that actually works

**Reply by rougetimelord** [reply]
OOOOOF

**Reply by rougetimelord** [reply]
I guess adding release/approval notes and license got added in v1.1.0-alpha.0? Probably not worth changing over yet https://github.com/wdzeng/firefox-addon/issues/2#issuecomment-2040386656

**Reply by cooljeanius** [reply]
how long do reviews usually take? Wonder if there should be a policy not to do a new release until the previous review has been completed...

**Reply by kheina** [reply]
> how long do reviews usually take?

previously it was really quick, but somewhere around v0.3.4 they updated their policy and every version requires human review, so it can take like 3-4 days now iirc

> Wonder if there should be a policy not to do a new release until the previous review has been completed...

I would agree

**Reply by rougetimelord** [reply]
> how long do reviews usually take? Wonder if there should be a policy not to do a new release until the previous review has been completed...

I agree! Sorry for adding 2 PRs a day lmao
Is there a way to get notified that a review completed?

**Reply by kheina** [reply]
I get emailed for it but beyond that idk. you can always check the addon page and see if the version is the latest one

**Reply by rougetimelord** [reply]
Ok, going to start forcing myself to do LLBs with new PRs being against those

**Comment by kheina** [comment]
> I guess adding release/approval notes and license got added in v1.1.0-alpha.0? Probably not worth changing over yet [wdzeng/firefox-addon#2 (comment)](https://github.com/wdzeng/firefox-addon/issues/2#issuecomment-2040386656)

I would be inclined to want to move over so I can stop checking the release version and updating it with content that's supposed to be added automatically. whole point of the release workflow is cause I've been ignoring this repo and I'm afraid if I forget the approval notes that they won't approve that version

**Reply by rougetimelord** [reply]
Okie dokie! Check #265

**Reply by kheina** [reply]
![Screenshot_20240418-093626](https://github.com/kheina-com/Blue-Blocker/assets/29378233/6aba4f0b-8fb4-4e1f-b63c-924831093b96)
there's no source code field on there is there

**Reply by cooljeanius** [reply]
darn it

**Comment by kheina** [comment]
https://github.com/marketplace/actions/firefox-addon-publish maybe? they support source code but not anything else

**Comment by kheina** [comment]
I think I'm going to clone https://github.com/browser-actions/release-firefox-addon and update it, or else try to add in a pr with fixes cause the fact that every one of these damn firefox release actions has bugs is really starting to piss me off

the source code license is hard coded to MIT, despite it accepting a license field:
https://github.com/browser-actions/release-firefox-addon/blob/9b26e1ce3122e6d70a7cffe3e36552786b84ff0c/src/index.ts#L88-L95

**Comment by kheina** [comment]
https://github.com/browser-actions/release-firefox-addon/pull/12 this should fix I think. who knows if they'll merge tho
